### ISSUE TITLE
`break or continue outside loop`

### ISSUE THREAD

**Issue by Moelf** [JuliaFolds2/OhMyThreads.jl#95]
I think this should be fine, but I guess the `@tasks` transformation made the Julia parsers unhappy:

```julia
function main(tree)
    btag_distribution = Hist1D(; bins = -2:100:2)
    @tasks for evt in tree
        @set scheduler=:static
        if !(evt.trigPassed_HLT_g35_tight_3j25_pf_ftf_PhysicsTLA_L1EM22VHI)
            continue
        end
        btag_score = D_b(evt)
        push!(btag_distribution, btag_score)
    end
end
```

**Comment by carstenbauer** [comment]
`@tasks` doesn't produce a for loop but maps to the higher order function API, `tforeach` in your case, which doesn't support `continue` or `break`. As for `continue`, we can maybe support it explicitly (in some cases), e.g. replace it by `return nothing`. `break` on the other hand doesn't make sense conceptually (I think). After all, the "loop" runs in parallel so it would lead to non deterministic behavior.

**Comment by Moelf** [comment]
Yeah continue works for `@threads` so it would be nice to have

**Comment by MasonProtter** [comment]
`return` inside of `@tasks` (or `tmapreduce`) does what `continue` does in a loop.

**Comment by carstenbauer** [comment]
I guess the question is whether we should replace `continue` → `return` automatically inside of `@tasks`.

**Comment by MasonProtter** [comment]
I'd lean towards no, because I don't want to be responsible for correctly parsing and understanding arbitrary user code. e.g. if they write
```
@tasks for x in v
    for i in 1:x
        if cond
            continue
        end
    end
end
```
then we'd actually want to leave that use of `continue` alone, lest we break their code in confusing and hard to debug ways.
___________________________



`break` is supported in Transducers.jl based stuff via their early termination interface. We probably could support it too, but it'd be a little tricky. It conceptually does make sense in some circumstances. For example, if you had a parallelized `any` function or a parallelized `all` function you'd want to `break` out of those as soon as you find a `true` or `false` respectively. 

The way Transducers does it is if one task signals that it encountered the early termination signal, then it basically just abandons the other tasks and never `fetch`es from them, signalling to them that they shouldn't spawn more tasks in the future, and then eagerly returns the final result.

**Comment by carstenbauer** [comment]
I've added a documentation label because we should at least mention this (`continue` in particular) somewhere in the docs/docstrings.

**Comment by carstenbauer** [comment]
I've created a [new issue](https://github.com/JuliaFolds2/OhMyThreads.jl/issues/96) for early termination support (e.g. `break` within `@tasks`) because the original example here is about `continue`.

**Comment by carstenbauer** [comment]
Another potential way to support `continue`:
```
julia> @tasks for i in 1:4
           for i in i # trivial loop to support continue
               if i > 2
                   continue
               end
               println(i)
           end
       end
2
1
```
